### PR TITLE
Reset options, notify user about backup creation

### PR DESCRIPTION
**Pull request by furszy**
Quick follow-up to first point of https://github.com/bitcoin-core/gui/pull/602#pullrequestreview-1002780997

### CODE REVIEW

**Comment by Rspigler** [comment]
`make check`

> make[4]: Leaving directory '/home/user/gui/src'
> Running tests: addrman_tests from test/addrman_tests.cpp
> /bin/bash: line 2: test/test_bitcoin: Permission denied
> make[3]: *** [Makefile:20843: test/addrman_tests.cpp.test] Error 1
> make[3]: Leaving directory '/home/user/gui/src'
> make[2]: *** [Makefile:18948: check-am] Error 2
> make[2]: Leaving directory '/home/user/gui/src'
> make[1]: *** [Makefile:18605: check-recursive] Error 1
> make[1]: Leaving directory '/home/user/gui/src'
> make: *** [Makefile:818: check-recursive] Error 1

**Comment by furszy** [comment]
@Rspigler issue is on your environment, this PR has no link with it.

"Permission denied" from bash usually means that the binary is not executable. Try cleaning your environment first (`git clean -dfx`) and recompile the sources (`make check`). (plus, if doesn't work, could try with `chmod +x src/test/test_bitcoin`).

**Comment by Rspigler** [comment]
tACK 52372cbfdb4d2cffd83f238cccab84fdc4cc6632
It writes `settings.json.bak`, but then continues to overwrite the file rather than create `settings1.json.bak`, settings2...., etc.

Tests do pass

**Review by shaavan — APPROVED** [review]
ACK 52372cbfdb4d2cffd83f238cccab84fdc4cc6632

- I agree with the idea of this PR. Displaying where the settings are getting backup lets the user know where to access them when needed.
- I successfully compiled this PR on Ubuntu 22.04 with `Qt version 5.15.3`.

**Screenshot:**

|Master|PR|
|---------|----|
|Screenshot from 2022-06-14 17-42-23|Screenshot from 2022-06-14 17-59-59|


**Observations:**
- Verified that the settings are correctly backed up on the displayed location.
- The latest custom settings overwrite the `settings.json.bak` file. Hence leading to backing up only the latest set of settings.

File: src/qt/optionsdialog.cpp [hunk 1]
```diff
@@ -278,14 +278,15 @@ void OptionsDialog::setOkButtonState(bool fState)
...
 void OptionsDialog::on_resetButton_clicked()
 {
-    if(model)
-    {
+    if (model) {
         // confirmation dialog
         QMessageBox::StandardButton btnRetVal = QMessageBox::question(this, tr("Confirm options reset"),
-            tr("Client restart required to activate changes.") + "<br><br>" + tr("Client will be shut down. Do you want to proceed?"),
+            tr("Client restart required to activate changes.") + "<br><br>" +
+            tr("Current settings will be backed up at \"%1\".").arg(GUIUtil::getDefaultDataDirectory()) + "<br><br>" +
```
[the remark this answers is not in the document]
Comment (furszy) [reply src/qt/optionsdialog.cpp:285]: yeah, good catch.
but one thing, will better provide client model access so we don't access `gArgs`  from the widget (at least in my head, the ideal architecture would be widget -> model -> interfaces -> backend).

**Comment by furszy** [comment]
Updated per ryanofsky feedback.

The flow uses now `m_client_model->dataDir()` instead of `GUIUtil::getDefaultDataDirectory()`.

File: src/qt/optionsdialog.cpp [hunk 2]
```diff
@@ -278,14 +278,15 @@ void OptionsDialog::setOkButtonState(bool fState)
...
+            tr("Client will be shut down. Do you want to proceed?"),
```
[the remark this answers is not in the document]
Comment (ryanofsky) [reply src/qt/optionsdialog.cpp:292]: re: https://github.com/bitcoin-core/gui/pull/617#discussion_r904773247

> Please add translator comments to these translatable strings. I've written some documentation on crafting translator comments here: https://github.com/bitcoin-core/gui-qml/blob/main/src/qml/doc/translator-comments.md
> 
> I can suggest the comments if you'd like

This seems worth following up on. It's not obvious to me what translator comments would be helpful here so maybe suggesting specific comments would be helpful

**Review by ryanofsky — APPROVED** [review]
Code review ACK ac4fb3bbbe207d0744201a9df8a971af06db5476, just fixing displayed backup directory since last review

**Comment by ryanofsky** [comment]
> suggesting the following translator comments:

This is great. Thanks!

**Comment by furszy** [comment]
Oh, sorry @jarolrod. I missed your comment there. Thanks for the suggestion there 👌🏼 .

I'm fine either way, could add it here under your authorship or you could create a follow-up PR for it.

-----------------------

Have to say that mixing source code written in c++ with comments for translators isn't convincing me. There should be another way, some external file/s where we could map every text being translated to its own translator comment so we don't end up adding external stuff to the sources (if something like this doesn't exist then maybe we should create it).

**Comment by jarolrod** [comment]
@furszy 
> Have to say that mixing source code written in c++ with comments for translators isn't convincing me. There should be another way, some external file/s where we could map every text being translated to its own translator comment so we don't end up adding external stuff to the sources (if something like this doesn't exist then maybe we should create it).

this is nothing new, we just follow Qt's process for [writing source code for translation](https://doc.qt.io/qt-5/i18n-source-translation.html). We've chosen to provide context to translators to help them with translating from english to a target language, and to do this we need to provide a [translator comment](https://doc.qt.io/qt-5/i18n-source-translation.html#translator-comments). I am not aware of any other way to provide translator context and nicely integrate it with transifex. Custom tooling seems like an unnecessary effort.

Translator comments are part of the burden of developing an application that needs to be localized; it is a worthwhile burden.

**Review by jarolrod — APPROVED** [review]
tACK ac4fb3bbbe207d0744201a9df8a971af06db5476

Will pick up translation fixup in a follow-up.

**Comment by furszy** [comment]
> @furszy
> 
> > Have to say that mixing source code written in c++ with comments for translators isn't convincing me. There should be another way, some external file/s where we could map every text being translated to its own translator comment so we don't end up adding external stuff to the sources (if something like this doesn't exist then maybe we should create it).
> 
> this is nothing new, we just follow Qt's process for [writing source code for translation](https://doc.qt.io/qt-5/i18n-source-translation.html). We've chosen to provide context to translators to help them with translating from english to a target language, and to do this we need to provide a [translator comment](https://doc.qt.io/qt-5/i18n-source-translation.html#translator-comments). I am not aware of any other way to provide translator context and nicely integrate it with transifex. Custom tooling seems like an unnecessary effort.
> 
> Translator comments are part of the burden of developing an application that needs to be localized; it is a worthwhile burden.

Yeah, I understood the rationale behind it (not the first time working with localized applications), I'm just not happy mixing code (which should be as clean as possible) with comments that should be properly crafted for non-technical people. We might end up having large cpp files full of non-technical comments that would make readability harder which is one of the main bug causes..

**Comment by jarolrod** [comment]
> We might end up having large cpp files full of non-technical comments that would make readability harder which is one of the main bug causes.

Won't be an issue with the switch to qml :)

**Pull request merged**